openedx/core/djangoapps/django_comment_common/comment_client/course.py

```diff
@@ -29,17 +31,20 @@ def get_course_commentable_counts(course_key: CourseKey) -> Dict[str, Dict[str,
             }
 
     """
-    url = f"{settings.PREFIX}/commentables/{course_key}/counts"
-    response = perform_request(
-        'get',
-        url,
-        metric_tags=[
-            f"course_key:{course_key}",
-            "function:get_course_commentable_counts",
-        ],
-        metric_action='commentable_stats.retrieve',
-    )
-    return response
+    if is_forum_v2_enabled(course_key):
+        commentable_stats = forum_api.get_commentables_stats(str(course_key))
+    else:
+        url = f"{settings.PREFIX}/commentables/{course_key}/counts"
+        commentable_stats = perform_request(
+            'get',
+            url,
+            metric_tags=[
+                f"course_key:{course_key}",
+                "function:get_course_commentable_counts",
+            ],
+            metric_action='commentable_stats.retrieve',
+        )
+    return commentable_stats
```

Comment:
the if-else condition seems quite redundant, maybe we can use a method decorator to achieve the same task. This will reduce git difference and make the mode easier to read.
```
def forum_v2_decorator(forum_v2_func):
    def decorator(func):
        def wrapper(course_key: CourseKey, *args, **kwargs):
            if is_forum_v2_enabled(course_key):
                return forum_v2_func(course_key, *args, **kwargs)
            else:
                return func(course_key, *args, **kwargs)
        return wrapper
    return decorator

@forum_v2_decorator(forum_api.get_commentables_stats)
def get_course_commentable_counts(course_key: CourseKey) -> Dict[str, Dict[str, int]]:
    url = f"{settings.PREFIX}/commentables/{course_key}/counts"
    commentable_stats = perform_request(
        'get',
        url,
        metric_tags=[
            f"course_key:{course_key}",
            "function:get_course_commentable_counts",
        ],
        metric_action='commentable_stats.retrieve',
    )
    return commentable_stats

```

Reply:
I will be avoiding this change for now, as the decorator method will become complex for handling native requests for all APIs. 
Note: As we will remove the else condition to V1 in future